### ISSUE TITLE
How to handle interpolated time outside range

### ISSUE THREAD

**Issue by joakim-hove** [Statoil/libecl#521]
This issue is about the `ecl_sum_get_interp()` family of functions. Assuming an eclipse summary case spans the time interval `[t0, t1]` if you call `xxx_get_interp()` with `t` &notin; `[t0,t1]` you will currently get an exception. 

As part of a desperate nexus fix here: https://github.com/Statoil/libecl/pull/506 and https://github.com/Statoil/libecl/pull/510 the `ecl_sum_resample_..` function was allowed to resample (i.e. interpolate) time values outside the interval - so right now the different interpolation functions are not consistent in this regard. We can:

1. Essentially revert https://github.com/Statoil/libecl/pull/506 and https://github.com/Statoil/libecl/pull/510 to recover the previous throw for &notin; behaviour.
2. We can let all interpolation accept &notin; - in that case rate values should be extrapolated with zero, all other values can be meaningfully extrapolated with first and last value for `t < t0` and `t > t1` respectively. 

I would vote for 2.

**Comment by markusdregi** [comment]
In 2, would we then interpolate outside the `[t0, t1]` interval as a step function, as in for `t1 + eps` any rate would be 0? Or with a function that extremely quickly decreases towards 0?

In general I guess the very best would be if we could support both and then it would be up to the user to argue whether one would like something like this or not?

**Comment by joakim-hove** [comment]
> In 2, would we then interpolate outside the [t0, t1] interval as a step function, as in for t1 + eps any rate would be 0? Or with a function that extremely quickly decreases towards 0?

Yes - step function.

> In general I guess the very best would be if we could support both and then it would be up to the user to argue whether one would like something like this or not?

Well - in a way we already do; the user can query for `t0` and `t1` and if the default behaviour of trivial extrapolation is not acceptable any alternative can be implemented. The point is that I *think* alternative 2 will have the highest satisfaction rate.

**Comment by markusdregi** [comment]
How about changing the API of `resample` as follows:

```python
def resample(self,
        new_case_name,
        time_points,
        lower_interpolation=False,
        upper_interpolation=False,
        ):
```

And then, if you do not toggle the `interpolation` arguments, you get the same old behaviour with failure. If you set them as `True`, you get the value of `start_time` if `time < start_time` and the value of `end_time` if `end_time < time` (as implemented in the above PR's). Finally, if you set `lower_interpolation` (or `upper_interpolation`) as a numeric value, i.e. `1`, you get `1` back for every `time < start_time`.

Then we support all of the scenarios you draw above, without enforcing any of these decisions on the user. Another advantage is that it is backwards compatible with the current behaviour :)

**Comment by joakim-hove** [comment]
OK - that will work. There are several entry points to the lowest-level interpolation functions, so although only one API function is changed there will be internal changes several places - that is OK.

> Finally, if you set lower_interpolation (or upper_interpolation) as a numeric value, i.e. 1, you get 1 back for every time < start_time.

That would have been elegant, but the `resample` function resamples a complete summary case, i.e. multiple vectors, and I don't think one non-trivial numerical value will be sensible for all of the vectors, so I suggest we limit the argument to `True|False`.

**Comment by xjules** [comment]
Solved by #534
